File: Frontend/src/interfaces/v12/userRole/userRole.ts
import { ModuleEnum } from "interfaces/enums/GlobalEnums";
import { guid } from "types/guid";
import { CreatedChangedEntity } from "../CreatedChangedEntity";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface UserRoleDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
  displayName: string;
  module: ModuleEnum[];
}

export interface UserRoleSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/UserRole/UserRole.cs
using System;
using Wms.Api.Entities;
using Wms.Api.Model;

namespace Wms.Api.Dto.UserRole
{
    public class UserRoleDetailsDto 
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public required string DisplayName { get; set; }
        public ICollection<ModuleEnum> Module { get; set; }
    }

    public class UserRoleCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class UserRoleSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}
